test/controllers/foreman_puppet/api/v2/environments_controller_test.rb

```diff
@@ -270,6 +270,18 @@ class EnvironmentsControllerTest < ActionController::TestCase
               assert_response :success
             end
 
+            test 'should render error message when invalid environment is set' do
+              env = 'invalid'
+              # assert_difference(-> { ForemanPuppet::Puppetclass.unscoped.count }, 0) do
```

Comment:
The debug code should be removed prior merge.

Reply:
Definitely. Right now, the test is failing, though. Any idea why?

Reply:
I don't have time to look here myself. @kamils-iRonin: Do you mind chiming in here?

Reply:
the test fails because the puppet proxy api returns mocked classes for any environment name, see:

https://github.com/theforeman/foreman_puppet/blob/3069d9022c41b69252932a5679d1735e28adb84f/test/controllers/foreman_puppet/api/v2/environments_controller_test.rb#L254

the test should pass after replacing this line with:

```ruby
ProxyAPI::Puppet.any_instance.stubs(:classes).with('invalid').returns({})
```

(tested with `develop` core branch)

Reply:
Thanks a lot, much appreciated @kamils-iRonin